while-loops/problem-set-while-loops.md

```diff
@@ -0,0 +1,849 @@
+import unittest
+import main
+from main import silly_sum
+
+num_src = None
+
+def input_ints(nums):
+    for num in nums:
+        yield num
+
+def input_int(*args):
+    return next(num_src)
+
+# inject into main
+main.input_int = input_int
+
+class TestPython1(unittest.TestCase):
+
+    def test_one(self):
+        global num_src
+        num_src = input_ints([1, 1, 2, 3, 5, 8, 0, 9])
+        self.assertEqual(20, silly_sum(), 'the provided input was 1, 1, 2, 3, 5, 8, 0, 9')
+
+    def test_two(self):
+        global num_src
+        num_src = input_ints([400, 400, 400, 400])
+        self.assertEqual(1200, silly_sum(), 'the provided input was 400, 400, 400, 400')
+
+    def test_three(self):
+        global num_src
+        num_src = input_ints([999, 1, 1])
+        self.assertEqual(1000, silly_sum(), 'the provided input was 999, 1, 1')
+```
+
+##### !end-tests
+
+#### !hint
+
+A possible interaction with a user could look like this:
+> Enter an integer (0 to stop): 999<br />
+> Enter an integer (0 to stop): 1<br />
+> The sum is: 1000
+
+Here's another possible example:
+> Enter an integer (0 to stop): 400<br />
+> Enter an integer (0 to stop): 400<br />
+> Enter an integer (0 to stop): 400<br />
+> The sum is: 1200
+
+This sum is halted by the user:
+> Enter an integer (0 to stop): 1<br />
+> Enter an integer (0 to stop): 2<br />
+> Enter an integer (0 to stop): 3<br />
+> Enter an integer (0 to stop): 0<br />
+> The sum is: 6
+
+Note that printing the final sum line is not a listed requirement, but feel free to add additional output to your implementation to help think through the problem. It will not affect the results of the tests. Even if you _do_ print out the final sum, be sure to also _return_ it!
+
+#### !end-hint
+
+#### !explanation
+
+An example of a working implementation:
+
+```python
+def silly_sum():
+    # accumulate the sum
+    my_sum = 0
+
+    # stores the most recent user input
+    # default to anything except 0 (think about why this is)
+    num = 1
+
+    # check our 2 conditions:
+    #   user input must not be 0
+    #   don't ask for more input once the sum reaches at least 1000
+    while num != 0 and my_sum < 1000:
+        # prompt for input
+        num = input_int('Enter an integer (0 to stop): ')
+
+        # add the input to the sum
+        my_sum += num
+
+    # return the sum
+    return my_sum
+```
+
+Read through this version and compare to yours. What's the same? What's different? Are there even more ways to approach this problem?
+
+<br />
+
+For some extra challenge, on your own, try modifying `silly_sum` so that you can pass in the stop value and maximum sum as parameters. This would allow us to say that 1 is the stop value instead of 0. How does this affect the code? Does anything else need to change? We could also say that the maximum sum is -1000! What changes does that cause?  
+
+<br />
+
+Simple changes can sometimes have complicated effects. Play around a bit and see what you can build!
```

Comment:
As mentioned with the other question-- You're adding an extension to this question to the explanation. This is fine, but I can imagine it being helpful to separate this and call it out as an extension/optional. I won't fight for this.

Reply:
Same as above.If people completely ignore this, fine. I just want there to be some framework for further consideration for those who might want to.